src/store/filestore.rs

```diff
@@ -1,207 +1,256 @@
 use crate::{
     error::{Kind, OrcaError, Result},
-    model::{from_yaml, to_yaml, Pod},
-    store::Store,
+    model::{from_yaml, to_yaml, Annotation, Pod},
+    util::get_type_name,
 };
 use colored::Colorize;
 use regex::Regex;
+use serde::{de::DeserializeOwned, Serialize};
 use std::{
     collections::BTreeMap,
     fs,
     path::{Path, PathBuf},
 };
-/// Support for a storage backend on a local filesystem directory.
+
+use super::{ItemKey, Store};
+
+/// Local storage system for orca items implmenting store
 #[derive(Debug)]
 pub struct LocalFileStore {
-    /// A local path to a directory where store will be located.
-    pub directory: PathBuf,
+    directory: PathBuf,
 }
 
 impl Store for LocalFileStore {
     fn save_pod(&self, pod: &Pod) -> Result<()> {
-        let class = "pod";
-        // Save the annotation file and throw and error if exist
-        Self::save_file(
-            &self.make_annotation_path(
-                class,
-                &pod.hash,
-                &pod.annotation.name,
-                &pod.annotation.version,
-            ),
-            &serde_yaml::to_string(&pod.annotation)?,
-            true,
-        )?;
-        // Save the pod and skip if it already exist, for the case of many annotation to a single pod
-        Self::save_file(
-            &self.make_spec_path(class, &pod.hash),
-            &to_yaml::<Pod>(pod)?,
-            false,
-        )?;
+        self.save_item(pod, &pod.hash, pod.annotation.as_ref())
+    }
 
-        Ok(())
+    fn load_pod(&self, item_key: &ItemKey) -> Result<Pod> {
+        self.load_item::<Pod>(item_key)
     }
 
-    fn load_pod(&self, name: &str, version: &str) -> Result<Pod> {
-        let class = "pod".to_owned();
+    /// Return Btree where key is column name and value is a vec of values
+    /// Are we okay with returning a bunch of strings? For now it works but later on like adding version
+    /// this will break...
+    #[expect(
+        clippy::unwrap_in_result,
+        reason = "It shouldn't failed as we added the keys ourselves at the start"
+    )]
+    #[expect(
+        clippy::unwrap_used,
+        reason = "It shouldn't failed as we added the keys ourselves at the start"
+    )]
+    fn list_pod(&self) -> Result<BTreeMap<String, Vec<String>>> {
```

Comment:
@Synicix Is going to change the return type to a `Vec<Struct>` and we'll revisit once the update is in.

Reply:
Done